port/unix/omrsysinfo.c

```diff
@@ -1705,6 +1707,22 @@ find_executable_name(struct OMRPortLibrary *portLibrary, char **result)
 		rc = 0;
 	}
 	return rc;
+#elif defined(OMR_BSD)
+	int proc[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+	size_t size = 0;
+	if (0 == sysctl(proc, sizeof(proc) / sizeof(proc[0]), NULL, &size, NULL, 0)) {
+		*result = portLibrary->mem_allocate_memory(portLibrary, size, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+	} else {
+
+		return -1;
+	}
```

Comment:
If you're going to be using multiple returns, I think this is better structured as:
```
if (0 != sysctl(...)) {
  return -1;
}

*result = portlibrary->mem_allocate(...);
```

Just like you did in the next call below.

And, don't forget to check if the new allocation worked.